### ISSUE TITLE
Add cursor next to selected item

### ISSUE THREAD

**Issue by beauwilliams** [camspiers/snap#40]
I currently have an issue where I can not see the selected item in the list, but seeing as nobody has mentioned it yet, might be just my setup.. however, this lead me to an enhancement suggestion, *a cursor icon next to current selection*

**such as telescope**
![image](https://i.ibb.co/zGwjcFJ/Screen-Shot-2021-06-22-at-3-57-29-pm.png)

If there was a cursor, this would be enough to solve my issue for now without having to rip apart my configs trying to figure out what the issue is

**fyi what issue is**
![image](https://i.ibb.co/XfQCDpS/ezgif-7-534ee7001ac3.gif)

**Comment by beauwilliams** [comment]
FYI I found the issue with the cursor not showing. All good now. Was my a plugin I am writing causing the issue.

**Comment by gegoune** [comment]
Please post the solution. I am not seeing the cursor either.

**Comment by beauwilliams** [comment]
> Please post the solution. I am not seeing the cursor either.

Yep so the issue I was having earlier was the gif below. I removed my own plugin [focus.nvim](https://github.com/beauwilliams/focus.nvim)

I believe it was likely related to auto window sizing on my end. It has a bug anyways I need to fix..

![image](https://i.ibb.co/XfQCDpS/ezgif-7-534ee7001ac3.gif)

**Comment by beauwilliams** [comment]
@camspiers do snap buffers have a filetype set? I would like to disable my plugin running on snap buffers

**Comment by camspiers** [comment]
They don't have a file type set, but if that would help I could set one (except for previews). The results and the input buffers are both of buftype prompt.

**Comment by camspiers** [comment]
With respect to a cursor, I think it's probably a good idea, I'll put it on my todo list.

**Comment by beauwilliams** [comment]
> They don't have a file type set, but if that would help I could set one (except for previews). The results and the input buffers are both of buftype prompt.

Actually I don't think its going to be necessary (at least myself for now)

I found the source of the issue being related to setting the below autocmds.

```lua
    { 'WinEnter', '*', 'setlocal cursorline' })
    { 'WinLeave', '*', 'setlocal nocursorline' })

```

Here is how I fixed the problem in my plugin. I suppose I should have been using BufEnter anyways

```lua
    { 'BufEnter', '*', 'setlocal cursorline' })
    { 'BufLeave', '*', 'setlocal nocursorline' })
```
